src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -0,0 +1,426 @@
+package com.snowflake.kafka.connector.internal.streaming;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.internal.Logging;
+import com.snowflake.kafka.connector.internal.SnowflakeConnectionService;
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import com.snowflake.kafka.connector.internal.SnowflakeSinkService;
+import com.snowflake.kafka.connector.internal.SnowflakeTelemetryService;
+import com.snowflake.kafka.connector.records.RecordService;
+import com.snowflake.kafka.connector.records.SnowflakeMetadataConfig;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is per task configuration. A task can be assigned multiple partitions. Major methods are
+ * startTask, insert, getOffset and close methods.
+ *
+ * <p>StartTask: Called when partitions are assigned. Responsible for generating the POJOs.
+ *
+ * <p>Insert and getOffset are called when {@link
+ * com.snowflake.kafka.connector.SnowflakeSinkTask#put(Collection)} and {@link
+ * com.snowflake.kafka.connector.SnowflakeSinkTask#preCommit(Map)} APIs are called.
+ *
+ * <p>This implementation of SinkService uses Streaming Snowpipe (Streaming Ingestion)
+ *
+ * <p>Hence this initializes the channel, opens, closes. The StreamingIngestChannel resides inside
+ * {@link TopicPartitionChannel} which is per partition.
+ */
+public class SnowflakeSinkServiceV2 implements SnowflakeSinkService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeSinkServiceV2.class);
+
+  private static String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
```

Comment:
Do we need a different name here for V2?

Reply:
From the initial thoughts, I don't think SnowflakeSinkServiceV2 is needed in this name. `KC_CLIENT ` seems a shorter version and we will also have connector name and taskID in streaming client name. 

Also, there isn't a similar name in V1. (Since there is no client name in ingest SDK for snowpipe)